src/services/ethereum/transaction.ts

```diff
@@ -171,7 +184,18 @@ export class TransactionExecutor implements ITransactionExecutor {
       } else {
         // tx might've been just confirmed on chain so let's check that as well
         try {
-          const provider = ethers.getDefaultProvider(ethers.providers.getNetwork(scheduled.chainId));
+          let provider: ethers.providers.BaseProvider;
```

Comment:
why repeat the whole logic here? you literally just implemented `getProvider` so why not call that instead? in general always avoid repeating yourself like that. this only leads to headaches